Track analytics of the standalone install button

kumar303 opened the pull request:
Fixes https://github.com/mozilla/addons-frontend/issues/3600

- is this the right way to do it?
- the `mozAddonManager` does it like this but only *after* a successful installation. I'm not sure if we need to set this event apart.
- the old site seems to be using a different analytics framework so I'm not sure if this matches what's going on there.

muffinresearch commented:
Would it be difficult to listen to the success event with the basic button if mozAddonManager is available? One issue with this is we're now using the non-mozAddonManager fallback button as the main button so this does need to work in both cases.

As you say the instrumentation is most useful based on the success events since the cancellation at the permissions prompt will result in no complete install.

If adding the success listening is better to be done when the changes to the button happen (as mentioned in #3600 we're still in need of a spec for that and that's not going to happen pre-release) maybe it makes sense to move forward with this but to refer to this separately as a new "install start" event and then we can add that into the instrumentation of both the switch and the new button to better understand when users bail on the install.

kumar303 commented:
@muffinresearch ok, I switched to an install started event so this is ready for another look. I also adjusted the InstallSwitch code to track an install started event just so we don't forget when we merge them.

> Would it be difficult to listen to the success event with the basic button if mozAddonManager is available

It's possible but I don't think it would be worth the effort. We need to merge the InstallSwitch functionality into the fallback button so we can do it later.

muffinresearch commented:
> It's possible but I don't think it would be worth the effort. We need to merge the InstallSwitch functionality into the fallback button so we can do it later.

Sounds good.

muffinresearch commented on tests/unit/core/TestInstallAddon.js:
```diff
@@ -935,6 +936,36 @@ describe(`${__filename}: withInstallHelpers`, () => {
+            category: INSTALL_STARTED_CATEGORY,
+            label: addon.name,
+          });
+          sinon.assert.calledOnce(fakeTracking.sendEvent);
```
Nice

kumar303 commented:
Good call. I added STR and QA notes to the issue
